### DISCUSSION TITLE
genmon web server intermittently offline, but host is up and I can still connect via ssh (WiFi)

### DISCUSSION THREAD

**Discussion by billpribyl** [jgyates/genmon#1118] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.02

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

New user here.

I'm connected to the genmon device using ssh over WiFi but when I try to connect to the web server it is (intermittently) unavailable for many minutes at a time. Previously, it has returned, but this time I haven't been able to connect for 30 minutes. Uptime on host Pi is 3 days. Connection seems fine; router shows WiFi connection to genmon has three bars.

I will most likely try rebooting the Pi, but wondering if there are less drastic troubleshooting steps to try first?

Not sure it matters, but we are in a utility outage situation at the moment. 

Next time the web server becomes available, I will send logs. Alternatively I could probably fetch them over ssh.

**Comment by billpribyl** [comment]
After some investigation, I noticed that genmonmaint.sh was running. It is roughly the time that I set for its weekly exercise. However, the generator was already running due to an outage. To restore access to the web server I tried doing a sudo kill on genserv and then launched it using nohup sudo /home/genmonpi/genmon/genenv/bin/python /home/genmonpi/genmon/genserv.py -c /etc/genmon/ &

It appeared that I was back in business...however, after a few minutes, I could no longer refresh any pages. So it's back to the same problem.

**Comment by jgyates** [comment]
genmonmaint.sh is a s script that is called for various reasons, but in most cases it is used by the user calling it directly. In only cases that the software will call genmonmaint.sh is to perform a software update or to backup the config files. Both of these are performed only at the users direction from the About page. The software will not call genmonmaint.sh without user interaction in some way.

I have seen this behavior before on several occasions and to date the only thing that has been found to be the source of the issue is one of two problems:

1) Genmon is exposed to the external internet via the local router config. this can and has caused issues with bots searching for open ports and trying a brute force attack on the web interface. In at least one occurrence a hacker has gained control of genmon and turned on a generator using remote commands.
2) Some routers use aggressive countermeasures on local web servers that do not have public certificates in an attempt to prevent malicious attacks on the network. Dialing back the security measures on the local network have shown that this will correct the issue.

 This thread has more info and some links to other similar issues:

https://github.com/jgyates/genmon/discussions/1116

If you do not expose genmon to the internet and you want to look into the router settings, if you post your router model I can take a look at some of the settings that may be applicable. I would like to be able to tell you which router and which setting, but so far I don't have enought data from users to be able to give any definitive data. I know I have seen this on some ASUS and NetGear routers.

**Comment by billpribyl** [comment]
Will send via PM as I don't want all my network configuration publicly available :-)

Edit: just as soon as I figure out how to send a PM on github...

**Reply by jgyates** [reply]
you can email me at generatormonitor.software@gmail.com
